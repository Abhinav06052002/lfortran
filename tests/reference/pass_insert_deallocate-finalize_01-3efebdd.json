{
    "basename": "pass_insert_deallocate-finalize_01-3efebdd",
    "cmd": "lfortran --pass=insert_deallocate --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/finalize_01.f90",
    "infile_hash": "3b4ed8b6df8d776bdd57e31aac947f9c2ee6dfc6d79ffb4aba9eb739",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "pass_insert_deallocate-finalize_01-3efebdd.stdout",
    "stdout_hash": "0a14333b99b9dcb6672e83205c8ecccc1065221aa985c4dab2685c2c",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}