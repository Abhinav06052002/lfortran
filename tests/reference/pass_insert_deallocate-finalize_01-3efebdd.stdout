(TranslationUnit
    (SymbolTable
        1
        {
            finalize_01:
                (Program
                    (SymbolTable
                        5
                        {
                            arr:
                                (Variable
                                    5
                                    arr
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Allocatable
                                        (Array
                                            (Integer 4)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            internal_sub:
                                (Function
                                    (SymbolTable
                                        6
                                        {
                                            arr_real:
                                                (Variable
                                                    6
                                                    arr_real
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Allocatable
                                                        (Array
                                                            (Real 4)
                                                            [(()
                                                            ())
                                                            (()
                                                            ())]
                                                            DescriptorArray
                                                        )
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            str:
                                                (Variable
                                                    6
                                                    str
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Allocatable
                                                        (String 1 -2 () DescriptorString)
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    internal_sub
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(ImplicitDeallocate
                                        [(Var 6 arr_real)
                                        (Var 6 str)]
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            ss:
                                (ExternalSymbol
                                    5
                                    ss
                                    2 ss
                                    finalize_01_mod
                                    []
                                    ss
                                    Public
                                ),
                            str:
                                (Variable
                                    5
                                    str
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Allocatable
                                        (String 1 -2 () DescriptorString)
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            test_01:
                                (ExternalSymbol
                                    5
                                    test_01
                                    2 test_01
                                    finalize_01_mod
                                    []
                                    test_01
                                    Public
                                )
                        })
                    finalize_01
                    [finalize_01_mod]
                    [(Allocate
                        [((Var 5 str)
                        []
                        (IntegerConstant 10 (Integer 4) Decimal)
                        ())]
                        ()
                        ()
                        ()
                    )
                    (SubroutineCall
                        5 ss
                        ()
                        []
                        ()
                    )
                    (SubroutineCall
                        5 internal_sub
                        ()
                        []
                        ()
                    )
                    (ImplicitDeallocate
                        [(Var 5 arr)
                        (Var 5 str)]
                    )
                    (Return)]
                ),
            finalize_01_mod:
                (Module
                    (SymbolTable
                        2
                        {
                            ss:
                                (Function
                                    (SymbolTable
                                        4
                                        {
                                            arr:
                                                (Variable
                                                    4
                                                    arr
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Allocatable
                                                        (Array
                                                            (Integer 4)
                                                            [(()
                                                            ())]
                                                            DescriptorArray
                                                        )
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    ss
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(Allocate
                                        [((Var 4 arr)
                                        [((IntegerConstant 1 (Integer 4) Decimal)
                                        (IntegerConstant 10 (Integer 4) Decimal))]
                                        ()
                                        ())]
                                        ()
                                        ()
                                        ()
                                    )
                                    (ImplicitDeallocate
                                        [(Var 4 arr)]
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            test_01:
                                (Function
                                    (SymbolTable
                                        3
                                        {
                                            str:
                                                (Variable
                                                    3
                                                    str
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Allocatable
                                                        (String 1 -2 () DescriptorString)
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    test_01
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(Allocate
                                        [((Var 3 str)
                                        []
                                        (IntegerConstant 10 (Integer 4) Decimal)
                                        ())]
                                        ()
                                        ()
                                        ()
                                    )
                                    (ImplicitDeallocate
                                        [(Var 3 str)]
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    finalize_01_mod
                    []
                    .false.
                    .false.
                )
        })
    []
)
